PriceQuoteIosTests/Salesman/SalesmanApiDataSourceTests.swift

```diff
@@ -0,0 +1,52 @@
+//
+//  SalesmanApiDataSourceTests.swift
+//  PriceQuoteIosTests
+//
+//  Created by Livia Vasconcelos on 27/09/20.
+//  Copyright © 2020 Livia Vasconcelos. All rights reserved.
+//
+
+import XCTest
+@testable import PriceQuoteIos
+
+class SalesmanApiDataSourceTests: XCTestCase {
+
+    func test_create_withCorrectFiels_requestSucceedes() {
+        let sut = SalesmanApiDataSourceImplMock()
+        let salesman = sut.salesman
+        let e = expectation(description: "Alamofire")
+        sut.status = .success
+        sut.create(salesman: salesman) { result in
+            result.onSuccess { response in
+                XCTAssertEqual(response.salesman?.name, salesman.name)
+            }
+            result.onFailed { error in
+                XCTFail("Fail to save salesman")
+            }
+            e.fulfill()
+        }
+        waitForExpectations(timeout: 5.0, handler: nil)
+    }
+}
+
+class SalesmanApiDataSourceImplMock: SalesmanApiDataSource {
```

Comment:
This is not testing that class at all, even if you are implementing the protocol, the code on the implementation is not being called :/

Reply:
Actually it is. I'm trying to code using the real implementation (like TDD), but the final code is using mock.